### ISSUE TITLE
[Feedback] Spending report

### ISSUE THREAD

**Issue by MatissJanis** [actualbudget/actual#1497]
Use this issue to post feedback about the new spending report  by @kyrias .

It is currently behind a feature flag - enable it as an experimental feature to test it.

<img width="1440" alt="Screenshot 2023-08-09 at 08 39 45" src="https://github.com/actualbudget/actual/assets/886567/378421af-747e-4c8b-b67e-aa482097ec88">

**Comment by Kidglove57** [comment]
I love the work that has been done! The thing I would really like to see is a “select all” or “deselect all” button rather than deselecting categories one by one.

**Comment by Crazypkr1099** [comment]
Hello @MatissJanis and @kyrias, I love your work you are doing with actualbudget.

I would like to suggest if you could check out another app called "Homebank" and view how they do their reports. I can post some pictures here as well, but if we could somehow implement what they have, I feel actualbudget would be amazing for reports. 

It gives some ideas on how to view data and export reports. I wish I was able to code, but unfortunately I can't, and not able to learn for a while.

Please let me know what you think, im not the creator or anyway affiliated with Homebank, just a general supporter of the reports, which im hoping actualbudget will be able to recreate and make them better.

Thanks!
![Screenshot 2023-08-11 192512](https://github.com/actualbudget/actual/assets/11566471/b4cffda9-1d06-467a-8686-5961d45ed8b1)
![Screenshot 2023-08-11 192630](https://github.com/actualbudget/actual/assets/11566471/750ac14d-b265-461b-be7b-a8eb697d3cb6)
![Screenshot 2023-08-11 192718](https://github.com/actualbudget/actual/assets/11566471/17e41a46-3e46-4327-a5c1-69590200f1f8)

**Comment by kyrias** [comment]
My opinion is essentially that if someone is willing to do the UI design and development work then I'd be happy to help actually implement the backend portion of it.  But I'm not really willing or able to do any real UI work myself.

**Comment by Crazypkr1099** [comment]
Do you mean images on how it should look? Or do you need someone to do the
front end coding portion. I might be able to help design it to fit into the
"actualbudget" look but I can't code unfortunately. I'm hoping to start
learning soon after 2 more exams of my actual job :)

**Comment by kyrias** [comment]
The latter, I'm afraid.  :)

**Comment by MatissJanis** [comment]
I think we can release the spending report in the next release. WDYT?

**Comment by kyrias** [comment]
I'm a bit conflicted.  On the one hand it _is_ useful as-is, but on the other hand the current Victory-based graphs leave a lot to be desired, like how the tooltip shows up in the middle of the _top_ (or _bottom_) of the category segment and the bar widths being weird on differently-wide screens.

**Comment by MatissJanis** [comment]
Are you @kyrias or anyone else working on those things?

Personally I don't like features being "experimental" for long periods of time. So either we cut the feature (if we feel like it's not in a suitable state and nobody is working on finishing it) or we release it (if we feel like it's in a stable state).

**Comment by kyrias** [comment]
For the tooltip I've tried but have no idea of how to fix the issue.  I've asked for assistance before, but no one else seemed to have any idea either.  The bar widths thing we cannot do anything about since it's inherent to how Victory works.

Switching from Victory to something else I could probably do, if a decision was made on that.

**Comment by MatissJanis** [comment]
>  if a decision was made on that

That's quite a difficult one to make. Generally I do think our current charting solution is sub-optimal. A POC here would be ideal.

**Comment by Crazypkr1099** [comment]
> Hidden categories are currently appearing in the report. Should we exclude hidden categories from the report?

What if we set the hidden reports as "off" and leave them to be available?

**Comment by youngcw** [comment]
> Hidden categories are currently appearing in the report. Should we exclude hidden categories from the report?

I think it would make more sense to have hidden categories and groups not show up unless its requested in some way.  Unchecked by default is an ok compromise though.

**Comment by kyrias** [comment]
> Another bug I noticed is that you cannot uncheck a group if it has no categories under it.

That's a good point, I think we should probably just hide groups without any visible categories.

**Comment by mlindgren** [comment]
I'd love to see a pie chart that breaks down spending by category as a percentage of total spending for the given period, e.g. within August I spent x% on rent/mortgage, y% on groceries, and so on. This is a pretty useful feature of services like Mint and Rocket Money. Unfortunately, I'm not a UI guy either, so I don't have much to contribute here. If I'm looking at the right page, though, it seems like Victory should support this: https://formidable.com/open-source/victory/gallery/custom-tooltip-labels

**Comment by marijnkieboom** [comment]
I don't know if this is actually expected behaviour or not, but when toggling the mode to blurr all numbers, the y-axis is still readable in the spending report. Wouldn't it be better if this were blurred as well?

**Comment by Spalfnaganalf1** [comment]
Is there anywhere within this new reporting or somewhere in the existing app where i can see total spending against a category over a certain period of time?  Like if i wanted to see how much i'd spent on groceries in the last 6 months as an example, without having to get out my calculator and add up the months?

**Comment by mlindgren** [comment]
@Spalfnaganalf1 I don't believe the app supports this, but if you're comfortable with SQL you can download a backup of your data, and then use a tool like [DB Browser for SQLite](https://sqlitebrowser.org/) to load the database and execute SQL queries against it yourself. To see the total amount for each category over a given time frame, you'd do something like:

```
SELECT categories.name, SUM(amount)
FROM transactions
INNER JOIN categories ON categories.id == transactions.category
WHERE transactions.date > 20220000 AND transactions.date < 20230000
GROUP BY transactions.category
```

Would definitely be nice to have native support for this :)

**Comment by Crazypkr1099** [comment]
> Is there anywhere within this new reporting or somewhere in the existing app where i can see total spending against a category over a certain period of time?  Like if i wanted to see how much i'd spent on groceries in the last 6 months as an example, without having to get out my calculator and add up the months?

@mlindgren 

We actually have a way to do this in app. If you select all accounts, you can type a filter in (for example groceries) and it will show everything labeled groceries. If you select the select all button, it gives you a running total 

This is the best way to do this as of right now, but reports/charting is a work in progress.

**Comment by MatissJanis** [comment]
👋 What's the status with this experimental feature? Are we good to launch it as a first party feature?

**Comment by kyrias** [comment]
The status from my point of view is essentially that I think that this might end up being essentially just a subset of the custom reports feature, and so might not be worthwhile keeping once that's in, so I've been holding off until that's finished up.

**Comment by MatissJanis** [comment]
That makes sense @kyrias ! Thanks for clarifying :)

**Comment by MatissJanis** [comment]
👋 Hey @kyrias !

Just wanted to touch-base with you about the spending reports. It's an amazing feature you developed, but it seems it hasn't gotten any activity in the past months.

We talked among the core-maintainers and we'd like to propose removing the spending report feature in favour of the custom reports that are currently being actively developed. Most of the spending report functionality should already be possible with custom reports and some of the missing features are coming soon.

Do you think this makes sense?

Lastly I'd like to say a big THANK YOU for kickstarting the reports page improvements. I think your work here inspired custom reports + gave it a good platform to build off of. So your work is very much appreciated! 🚀

**Comment by kyrias** [comment]
@MatissJanis 

I'd been planning on opening a PR removing it once the custom reports are finished up, but since they're both experimental features and the custom reports feature is mostly there I guess I don't have any problems with it being removed now.  :)

**Comment by MatissJanis** [comment]
Sounds great! Would you prefer to do the cleanup yourself? I am happy to do it on my end too. Up to you :)

Reason I'm trying to get this done sooner rather than later is that it will unblock removing `victory` dependency. A small win for the codebase :)

**Comment by MatissJanis** [comment]
Removed in https://github.com/actualbudget/actual/pull/2344
